aaelf64/aaelf64.rst

```diff
@@ -1970,6 +1970,98 @@ Custom PLTs
 
 The information on custom PLTs has moved to [SYSVABI64_].
 
+.. raw:: pdf
+
+   PageBreak
+
+
+.. _appendix:
+
+Appendix
+========
+
+AArch64 Veneer Types Recognized by Binary Analysis Tools
+------------------------------------------------------------
+
+As described in `Call and Jump relocations`_, the linker may insert veneers (also referred to as thunks, stubs or trampolines) to implement call and jump relocations. This section defines the commonly used types and instruction sequences for such veneers on AArch64, enabling better recognition by binary analysis tools.
+
+Toolchains are encouraged to follow these patterns to ensure veneers can be reliably recognized during binary analysis. These pattern sets are intended to remain open, and toolchains may introduce new veneer forms.
+
+Additionally, if veneers emit static relocations, they become indistinguishable from regular code. In such cases, binary analysis tools should rely on relocation information to identify veneer targets.
+
+__AArch64AbsLongThunk_ Long-Range Veneers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A 64-bit absolute target address is loaded from a literal pool into a register, and an indirect branch is used to transfer control to the target::
+
+  <caller>:
+    B/BL __AArch64AbsLongThunk_<target>
+
+  __AArch64AbsLongThunk_<target>:
+    LDR X16, =<target>
+    BR  X16
```

Comment:
Is there an expectation that the literal pool follows the instruction sequence?

Reply:
I've occasionally seen people with embedded systems (not set up MMU yet) have a boot loader in low memory that jumps to a high address.

I suspect that if we default to using the ADRP thunk (4 GiB) universally, either using it within range and falling back to the longer sequence if out of range, or for lld at least defaulting --pic-veneer to True which will force use of the ADRP thunk.